### PR TITLE
Resolve #554 Do not output a full stack trace for a missing table

### LINKED ISSUE #554: 4.x - Fail nicely when missing a table
## What you did
`bin/cake bake template --theme CakeBootstrap --prefix Admin Questiojs`

## What happened?
`Exception: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'cakephp4testing.questiojs' doesn't exist in [/Users/davidyell/Sites/CakePHP4/vendor/cakephp/cakephp/src/Database/Schema/Collection.php, line 134]`

## What you expected to happen
I would think this could output a more concise error message, rather than a PDO exception.

### PR DESCRIPTION
**Pull request by davidyell**
I have made a start on trying to [reduce the amount of noise](https://i.imgur.com/qtpM2ar.png) generated by an error when baking templates.

It seems that when the `getPrimaryKey()` method is called, this causes a chain reaction, because the schema doesn't exist etc.

I'm open to suggestions to improve the Exception handling, as I can see some instances where a stack trace might be helpful. Perhaps the error could be logged instead?

Thoughts welcome and I can update the pull request. 

Thanks!

### CODE REVIEW

File: src/Command/TemplateCommand.php [hunk 1]
```diff
@@ -270,13 +270,18 @@ protected function _loadController(): array
...
+            $primaryKey = (array)$modelObject->getPrimaryKey();
+            $displayField = $modelObject->getDisplayField();
+            $singularVar = $this->_singularName($this->controllerName);
+            $singularHumanName = $this->_singularHumanName($this->controllerName);
+            $schema = $modelObject->getSchema();
+            $fields = $schema->columns();
+            $modelClass = $this->modelName;
+        } catch (\Exception $exception) {
+            echo $exception->getMessage() . "\n";
+            exit;
```
Comment (dereuromark) [inline src/Command/TemplateCommand.php]: exit makes the whole thing not testable. also no error code but success code 0 here.
so this is currently worse than before.

Comment (davidyell) [reply src/Command/TemplateCommand.php]: Please can you provide constructive feedback. You've provided no requested changes or route to move forward.

Comment (dereuromark) [reply src/Command/TemplateCommand.php]: It was - you just have to either output a 1 exit code (see status code constants in Command class) - or throw some "AbortException", I suspect the code also offers some abort() convenience wrapper for this.

Comment (davidyell) [reply src/Command/TemplateCommand.php]: Great, thanks. That's much better than 'this is currently worse than before'

Comment (markstory) [reply src/Command/TemplateCommand.php]: You can use `$this->abort()` here.

File: src/Command/TemplateCommand.php [hunk 2]
```diff
@@ -270,13 +270,18 @@ protected function _loadController(): array
...
+        try {
+            $primaryKey = (array)$modelObject->getPrimaryKey();
+            $displayField = $modelObject->getDisplayField();
+            $singularVar = $this->_singularName($this->controllerName);
+            $singularHumanName = $this->_singularHumanName($this->controllerName);
+            $schema = $modelObject->getSchema();
+            $fields = $schema->columns();
+            $modelClass = $this->modelName;
+        } catch (\Exception $exception) {
+            echo $exception->getMessage() . "\n";
```
Comment (markstory) [inline src/Command/TemplateCommand.php]: You should use `$io->error()` here. You'd need to pass the `ConsoleIo` into `_loadController()` though. `echo` can't be caught by the test harnesses.

**Comment by markstory** [comment]
> I'm open to suggestions to improve the Exception handling, as I can see some instances where a stack trace might be helpful. Perhaps the error could be logged instead?

I have considered removing the stack trace from the default console exception handling entirely. I agree it can be a bit noisy and hide the error message. The trace could be output if the `--verbose` flag was set though.

**Comment by davidyell** [comment]
Just going to add a test for the new output

**Comment by davidyell** [comment]
Let me know if this needs any more tweaks. Thanks!

**Pull request merged**